examples/calendar/backend/services/auth/controller.ts

```diff
@@ -0,0 +1,110 @@
+import { recoverPersonalSignature } from "eth-sig-util";
+import { bufferToHex } from "ethereumjs-util";
+import { NextFunction, Request, Response } from "express";
+import jwt from "jsonwebtoken";
+
+import { config } from "../../config";
+import { User } from "../../models/user.model";
+
+export const create = (req: Request, res: Response, next: NextFunction) => {
+  const { signature, publicAddress } = req.body;
+  if (!signature || !publicAddress)
+    return res
+      .status(400)
+      .send({ error: "Request should have signature and publicAddress" });
+
+  return (
+    User.findOne({ where: { publicAddress } })
+      ////////////////////////////////////////////////////
+      // Step 1: Get the user with the given publicAddress
+      ////////////////////////////////////////////////////
+      .then((user: User | null) => {
+        if (!user) {
+          res.status(401).send({
+            error: `User with publicAddress ${publicAddress} is not found in database`,
+          });
+
+          return null;
+        }
+
+        return user;
+      })
+      ////////////////////////////////////////////////////
+      // Step 2: Verify digital signature
+      ////////////////////////////////////////////////////
+      .then((user: User | null) => {
+        if (!(user instanceof User)) {
```

Comment:
we can add this check at line 22.

Reply:
I am just using a template starter code (https://github.com/amaurym/login-with-metamask-demo) to implement this.